### PR TITLE
pubsub: document intended use of Nack

### PR DESCRIPTION
**Pull request by zombiezen**
Follow-up from an offline discussion about Pub/Sub where I learned how to use Nack properly from @vangent.

### CODE REVIEW

File: pubsub/pubsub.go [hunk 1]
```diff
@@ -164,8 +164,14 @@ func (m *Message) Ack() {
 }
 
 // Nack tells the server that this Message was not processed and should be
-// redelivered. It returns immediately, but the actual nack is sent in the
-// background, and is not guaranteed to succeed.
+// redelivered (negative acknowledgement). It returns immediately, but the
+// actual nack is sent in the background, and is not guaranteed to succeed.
+//
+// Nack is a performance optimization for retrying transient failures: it
```
Comment (jba) [inline pubsub/pubsub.go]: "transient failures. It"

File: pubsub/pubsub.go [hunk 2]
```diff
@@ -164,8 +164,14 @@ func (m *Message) Ack() {
 }
 
 // Nack tells the server that this Message was not processed and should be
```
Comment (jba) [inline pubsub/pubsub.go]: How about

   Nack (short for negative acknowledgment) is a ...

Also note the omitted "e" in acknowledgment; that is the preferred American spelling. https://grammarist.com/spelling/acknowledgment-acknowledgement

Comment (zombiezen) [reply pubsub/pubsub.go]: Done.

File: pubsub/pubsub.go (hunk 1 above)
Comment (zombiezen) [reply pubsub/pubsub.go]: Not digging the multiple colon-delimited phrases? 😆 Done.

**Pull request merged**